src/ast/passes/semantic_analyser.cpp

```diff
@@ -595,6 +595,11 @@ static const std::map<std::string, call_spec> CALL_SPEC = {
       .max_args=1,
       .discard_ret_warn = true },
   },
+  { "socket_cookie",
+    { .min_args=1,
+      .max_args=1,
+      .arg_types={
+        arg_type_spec{ .type=Type::pointer } } } },
```

Comment:
nit: Add a comment like the one on line 495 to document the pointer type: `struct sock *`.